presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -154,4 +189,43 @@ private RuntimeException propagate(Exception e)
         throwIfUnchecked(classified);
         throw new RuntimeException(classified);
     }
+
+    private long getRetryDelayInMs(RetryStatus retryStatus)
```

Comment:
Good refactoring extracting this and the next method. It not only allows reuse, but also increases readability of the calling method.